### ISSUE TITLE
Rules for naming Variables are confusing

### ISSUE THREAD

**Issue by microbit-mark** [microsoft/pxt-microbit#2047]
from micro:bit support: 22167

Some feedback from a user:
[microbit-VARIABLETests.hex.zip](https://github.com/microsoft/pxt-microbit/files/3176377/microbit-VARIABLETests.hex.zip)

"JavaScript has only a few rules for variable names:

- The first character must be a letter or an underscore (_). You can't use a number as the first character.

 In BLOCK mode: `1stName` is allowed. Then in JS it is changed to `_1stName` and then the Block mode version is modified !!!

 - The rest of the variable name can include any letter, any number, or the underscore. You can't use any other characters, including spaces, symbols, and punctuation marks.

In BLOCK Mode: `First Name` is allowed. Then in JS it is changed to `First_Name`. But in Block mode it is still `First Name` !!!
In BLOCK Mode: `First.Name` is allowed. Then in JS it is changed to `FirstName`. But in Block mode it is still `First.Name` !!!
In BLOCK Mode: `First/Name` is allowed. Then in JS it is changed to `FirstName`, which conflicts with the previous `FirstName`; so the new one is changed to `FirstName2`. At the same time the previous `First.Name` in BLOCK mode is also now changed to `FirstName` !!!

- As with the rest of JavaScript, variable names are case sensitive. That is, a variable named Interest_Rate is treated as an entirely different variable than one named interest_rate.

Not in MakeCode

- There's no limit to the length of the variable name.

This one is actually observed. But I find it the most absurd one to follow, because it makes BLOCK coding totally unusable !!! If you are going to do any custom rules, then restrict this one.

- You can't use one of JavaScript's reserved words as a variable name.

In BLOCK mode reserved words are allowed: eg: `void`. In JS mode it is changed to` _void`, but it is still `void` in Block mode !!!

In short, variable name treatment is very confusing, specially for little kids unaware of the nuances of JS as a language.

A much SAFER approach would have been to enforce the naming rules up front and in the process teach the kids the naming rules without allowing weirdness to creep into the code."

**Comment by microbit-mark** [comment]
@abchatra can you take a look at this?

**Comment by pelikhan** [comment]
Kids unaware of JavaScript should not be burdened with learning the syntax requirements of JS variable names. Thus, we allow “natural language” names, including spaces and reserved keywords.
The same applies to function names or in general any of blocks do not match 1-to-1 the JavaScript syntax.

**Comment by abchatra** [comment]
Kids don't understand javascript. Design choice is give freedom to name whatever. There is no rules. Blocks language is not same as javascript. There is no scoping either. 

This is by design.
